include/cosim/slave.hpp

```diff
@@ -221,6 +221,60 @@ class slave
     virtual void set_string_variables(
         gsl::span<const value_reference> variables,
         gsl::span<const std::string> values) = 0;
+
+    /// Result type for `get_variables()`.
+    struct variable_values
+    {
+        /// Real variable values.
+        std::vector<double> real;
+
+        /// Integer variable values.
+        std::vector<int> integer;
+
+        /// Boolean variable values.
+        boost::container::vector<bool> boolean;
+
+        /// String variable values.
+        std::vector<std::string> string;
+    };
+
+    variable_values get_variables(
+        gsl::span<const value_reference> real_variables,
+        gsl::span<const value_reference> integer_variables,
+        gsl::span<const value_reference> boolean_variables,
+        gsl::span<const value_reference> string_variables) const
+    {
+
+        variable_values values;
```

Comment:
I think we should let the client to decide to whether to re-use this variable for improving performance. For this I made #687.